botocore/regions.py

```diff
@@ -63,8 +64,13 @@ def construct_endpoint(self, service_name, region_name, **kwargs):
                                          region_name, **kwargs)
 
         if endpoint is None:
-            raise UnknownEndpointError(service_name=service_name,
-                                       region_name=region_name)
+            if region_name is None:
+                # Raise a more specific error message that will give
+                # better guidance to the user what needs to happen.
+                raise NoRegionError()
+            else:
+                raise UnknownEndpointError(service_name=service_name,
```

Comment:
Out of curisosity, how does the ``UnknownEndpointError`` get thrown now? It may be good to add a test for it if there is not one already.